scheduler/pkg/kafka/gateway/consumer.go

```diff
@@ -169,6 +174,22 @@ func collectHeaders(kheaders []kafka.Header) map[string]string {
 	return headers
 }
 
+func (kc *InferKafkaConsumer) Produce(msg *kafka.Message, deliveryChan chan kafka.Event) error {
+	kc.producerMu.RLock()
+	defer kc.producerMu.RUnlock()
+	if kc.running.Load() {
+		return kc.producer.Produce(msg, deliveryChan)
+	} else {
+		return fmt.Errorf("The infer producer %s is no longer running", kc.producer.String())
+	}
+}
+
+func (kc *InferKafkaConsumer) closeProducer() {
+	kc.producerMu.Lock()
+	defer kc.producerMu.Unlock()
+	kc.producer.Close()
```

Comment:
:question: Is it allowed to close a producer (or consumer) multiple times?  If not, we should be checking here if it's already closed by the time this method takes the lock.

Reply:
In the interests of being defensive I think it's generally good to guard against this in case someone adds a bug in the future - this is probably a fairly tricky bug to catch if it does happen.  Not a blocker for this PR mind - happy to create a little follow-up that also renames the method receivers from `kc` -> `kh`